app/src/main/java/com/cmput301f20t21/bookfriends/ui/profile/ProfileEditDialog.java

```diff
@@ -56,6 +71,28 @@ public void onClick(View v) {
                     listener.onEdit(inputEmail, inputPhone);
                     getDialog().dismiss();
                 }
+                //update email authentication
+                FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
+                user.updateEmail(inputEmail)
+                        .addOnCompleteListener(new OnCompleteListener<Void>() {
+                            @Override
+                            public void onComplete(@NonNull Task<Void> task) {
+                                if (task.isSuccessful()) {
+                                    Log.d(TAG, "User email address updated.");
+                                }
+                            }
+                        });
```

Comment:
suggestion: extract this to a new method inside the [profile view model](https://github.com/CMPUT301F20T21/Book_Friends/blob/master/app/src/main/java/com/cmput301f20t21/bookfriends/ui/profile/ProfileViewModel.java) called `updateCurrentUserEmail`. use that function to do all the thing about updating emails (including the code below)

Reply:
we might also want to abstract updateEmail in AuthService. We don't want to expose FirebaseUser class throughout the app because it has too much power (example: we could delete user by doing user.delete()).

Reply:
For updateEmail in AuthService, so you mean I need to implement a method in updateEmail in User entity?

Reply:
I think Khang means move this section to AuthService class and have a new method called updateEmail(). Anything directly connecting to firebase/firestore should be done in the service classes.